filters/AssignFilter.cpp

```diff
@@ -123,13 +123,20 @@ void AssignFilter::addArgs(ProgramArgs& args)
 {
     args.add("assignment", "Values to assign to dimensions based on range.",
         m_assignments);
+    args.add("condition", "Condition for assignment based on range.",
+        m_condition);
 }
 
 
 void AssignFilter::prepared(PointTableRef table)
 {
     PointLayoutPtr layout(table.layout());
 
+    m_condition.m_id = layout->findDim(m_condition.m_name);
+    if (m_condition.m_id == Dimension::Id::Unknown)
```

Comment:
If no condition is provided, then `filters.assign` should continue to function as it always has, applying the assignment to **all points**, instead of throwing an error.